### ISSUE TITLE
Flutter has suddenly stopped working on Vscode

### ISSUE THREAD

**Issue by johnareid54** [Dart-Code/Dart-Code#2959]
**Describe the bug**
Using VS code this morning, when all of a sudden it stopped working! It reported errors that starting with 2.7 dart required a configuration file - the folder - dart-tool exists but it is empty.

following error when i run flutter packages pub run build_runner build

Failed to precompile build_runner:build_runner:
.dart_tool/package_config.json: Error: Problem in packages configuration file: Unexpected end of input.

OK i thought perhaps there has been an automatic update so I tried to create a new flutter project, the command did not show up in the menu list as normal but when I found it and tried to create a new project it failed, only getting as far as creating the project folder and an empty file called flutter.create.

**To Reproduce**

**Expected behavior**

**Screenshots**

**Versions (please complete the following information):**
 - VS Code version:
Version: 1.51.1
Commit: e5a624b788d92b8d34d1392e4c4d9789406efe8f
Date: 2020-11-11T01:11:34.018Z
Electron: 9.3.3
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0
 - Dart extension version:  3.16.0
 - Dart/Flutter SDK version:  3.16.0

Flutter doctor:

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 1.23.0-18.1.pre, on Mac OS X 10.15.6 19G73 x86_64, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
 
[!] Xcode - develop for iOS and macOS (Xcode 11.6)
    ✗ Xcode requires additional components to be installed in order to run.
      Launch Xcode and install additional required components when prompted or run:
        sudo xcodebuild -runFirstLaunch
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.0)
[!] Android Studio
    ✗ Android Studio not found at /Users/john/Library/Android/sdk/Contents
[✓] VS Code (version 1.51.1)
[✓] Connected device (2 available)

**Comment by DanTup** [comment]
Does the file `.dart_tool/package_config.json` get created if you run the **Get Packages** command? Pub should create it during package download, and then it's used for package resolution. You should have that file if the packages have been downloaded using a recently new version of pub.

**Comment by johnareid54** [comment]
Thanks for the quick response!

I think the problem is hardware-based. The hard disk was wiping hidden files off the drive, it was just that I was using flutter and Vscode at that time, so I assumed it was just a dart issue. The hard disk did not report any errors but was not available when I tried to do a restore.  I had to erase the disk and go back to a safe restore and now in the process of installing all the software again.  Slight chance it could have been a virus!
